api/revisions/serializers.py

```diff
@@ -0,0 +1,118 @@
+from api.base.serializers import (
+    JSONAPISerializer,
+    LinksField,
+    RelationshipField,
+    VersionedDateTimeField,
+)
+
+from rest_framework import serializers as ser
+from api.base.utils import absolute_reverse
+from osf.models.schema_responses import SchemaResponses
+from osf.models import Registration
+from rest_framework import exceptions
+from django.utils import timezone
+
+class SchemaResponsesSerializer(JSONAPISerializer):
+    id = ser.CharField(required=False, source='_id', read_only=True)
+    responses = ser.JSONField(required=False)
+    date_created = VersionedDateTimeField(source='created')
+    date_modified = VersionedDateTimeField(source='modified')
+    revision_justification = ser.CharField(source='justification')
+    revision_response = ser.JSONField(source='all_responses')
+    reviews_state = ser.ChoiceField(choices=['revision_in_progress', 'revision_pending_admin_approval', 'revision_pending_moderation', 'approved'], read_only=True)
+    is_pending_current_user_approval = ser.SerializerMethodField()
+
+    links = LinksField(
+        {
+            'self': 'get_absolute_url',
+        },
+    )
+
+    registration = RelationshipField(
+        related_view='registrations:registration-detail',
+        related_view_kwargs={'node_id': '<parent._id>'},
+        read_only=True,
+    )
+
+    registration_schema = RelationshipField(
```

Comment:
We had proposed having this as a FK directly on the SchemaResponses, and I think that's the right call. We'll want access to it for creating/validating ResponseBlocks, so it makes sense for the SchemaResponses to be aware of it instead of inheriting it from the parent. Also, that lets us attach SchemaResponses to entities that may not already have a "schema" field defined.

Reply:
Added schema to the SchemaResponses model, so this can just point to schema._id now instead of parent.registered_schema_id